sched/irq/irq_attach_wqueue.c

```diff
@@ -0,0 +1,219 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <errno.h>
+#include <stdio.h>
+
+#include <nuttx/irq.h>
+#include <nuttx/wqueue.h>
+#include <debug.h>
+
+#include "irq/irq.h"
+#include "sched/sched.h"
+
+/****************************************************************************
+ * Privte Types
+ ****************************************************************************/
+
+/* This is the type of the list of interrupt handlers, one for each IRQ.
+ * This type provided all of the information necessary to irq_dispatch to
+ * transfer control to interrupt handlers after the occurrence of an
+ * interrupt.
+ */
+
+struct irq_work_info_s
+{
+  xcpt_t handler;     /* Address of the interrupt handler */
+  xcpt_t isrwork;     /* Address of the interrupt worked handler */
+  FAR void *arg;      /* The argument provided to the interrupt handler. */
+  int irq;            /* Irq id */
+  struct work_s work; /* Interrupt work to the wq */
+
+  FAR struct kwork_wqueue_s *wqueue;   /* Work queue. */
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_MINIMAL_VECTORTABLE
+static struct irq_work_info_s
+g_irq_work_vector[CONFIG_ARCH_NUSER_INTERRUPTS];
+#else
+static struct irq_work_info_s g_irq_work_vector[NR_IRQS];
+#endif
+
+static mutex_t g_irq_wqueue_lock = NXMUTEX_INITIALIZER;
+static FAR struct kwork_wqueue_s *g_irq_wqueue[CONFIG_IRQ_NWORKS];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static
+inline_function FAR struct kwork_wqueue_s *irq_get_wqueue(int priority)
+{
+  FAR struct kwork_wqueue_s *queue;
+  int wqueue_priority;
+  int i;
+
+  nxmutex_lock(&g_irq_wqueue_lock);
+  for (i = 0; g_irq_wqueue[i] != NULL && i < CONFIG_IRQ_NWORKS; i++)
+    {
+      wqueue_priority = work_queue_priority_wq(g_irq_wqueue[i]);
+      DEBUGASSERT(wqueue_priority >= SCHED_PRIORITY_MIN &&
+                  wqueue_priority <= SCHED_PRIORITY_MAX);
+
+      if (wqueue_priority == priority)
+        {
+          nxmutex_unlock(&g_irq_wqueue_lock);
+          return g_irq_wqueue[i];
+        }
+    }
+
+  DEBUGASSERT(i < CONFIG_IRQ_NWORKS);
+
+  queue = work_queue_create("isrwork", priority,
+                            CONFIG_IRQ_WORK_STACKSIZE, 1);
+
+  g_irq_wqueue[i] = queue;
+  nxmutex_unlock(&g_irq_wqueue_lock);
+  return queue;
+}
+
+/* Default interrupt handler for worked interrupts.
+ * Useful for oneshot interrupts.
+ */
+
+static void irq_work_handler(FAR void *arg)
+{
+  FAR struct irq_work_info_s *info = arg;
+
+  info->isrwork(info->irq, NULL, info->arg);
+}
+
+static int irq_default_handler(int irq, FAR void *regs, FAR void *arg)
+{
+  FAR struct irq_work_info_s *info;
```

Comment:
```suggestion
  FAR struct irq_work_info_s *info = arg;
```
and remove ndx, vector